[Jessica Jacelyn] iP

jessicajacelyn opened the pull request:
### jBot 👾

> "jBot is here to remind you of your ToDos!" 📝

jBot is a bot to frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- feels like you are talking to someone
- perfect for fast typer like you

All you need to do is,
- download it from [here](https://github.com/jessicajacelyn/ip/releases/download/v0.2/jBot.jar).
- double-click it.
- add your tasks.
- let it manage your tasks for you 😉

and best of all ... it is FREE !!

Features:
- [ ] Managing ToDos
- [ ]   Managing Deadlines
- [ ]   Keep track of future events
- [ ]   Mark todos when they are completed
- [ ]   Reminders (coming soon) 🤓

If you are a Java programmer, you can use it to practice Java too!!
Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

jessicajacelyn committed:
Level 1 completed

jessicajacelyn committed:
Level 2 completed

jessicajacelyn committed:
Level 3 completed

jessicajacelyn committed:
Level 4 completed

jessicajacelyn committed:
Level-5 completed

jessicajacelyn committed:
Level-6 completed

jessicajacelyn committed:
Retrieve and append task to file

jessicajacelyn committed:
Complete level-8

jessicajacelyn committed:
Comple More-OOP increment

jessicajacelyn committed:
Package classes into Duke package

jessicajacelyn committed:
Add JUnit tests

jessicajacelyn committed:
Add JavaDoc headers to methods

jessicajacelyn committed:
Delete redundant files

jessicajacelyn committed:
Edit code to follow coding standard

jessicajacelyn committed:
Complete Level-9

jessicajacelyn committed:
Merge branch 'branch-A-CodingStandard'

jessicajacelyn committed:
Merge branch 'branch-Level-9'

jessicajacelyn commented:
### Dootoo 👾

> _"Dootoo is here to remind you of your ToDos!"_ 📝


Dootoo is a bot to frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- feels like you are talking to someone
- suitable for fast typer like you

All you need to do is,
1. download it from [here](https://github.com/jessicajacelyn/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

and best of all ... it is **FREE** !!

Features:
- [ ] Managing ToDos
- [ ] Managing Deadlines
- [ ] Keep track of future events
- [ ] Mark todos when they are completed
- [ ] Reminders (coming soon) 🤓

If you are a Java programmer, you can use it to practice Java too!!
Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

xzzz3 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,148 @@
+
+        // "bye" to end the program
+        switch (words[0]) {
+            case "bye" -> System.out.println("    Bye!! See you again soon!!");
```
If I remember correctly, the indentation level should be the same for switch and case statements - you can double check the 2103T standards!

xzzz3 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,48 @@
+
+import java.time.LocalDateTime;
+
+public class Task {
```
Should `Task` be an abstract class, considering that it shouldn't be initialized on its own? I noticed that you have overwritten a few methods like mark() in your child-classes, maybe those can be the abstract methods!

xzzz3 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,63 @@
+    ArrayList<Task> tasks;
+    public int items;
+
+    public TaskList(ArrayList<Task> t, int number) {
```
Perhaps a more descriptive variable name here? For instance, task and items, which corresponds to your overall implementation of this class! I noticed the variables in the other methods in this class may be changed in a similar way as well.

xzzz3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,36 @@
+    private static Storage storage;
+    private static TaskList tasks;
+
+    public Duke(String filePath) throws IOException {
+        Ui ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load(), storage.getNumberOfTasks());
+        } catch (IOException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
```
I like this Duke class initialization!

xzzz3 reviewed (commented):
Overall I think the code is great, with a few minor improvements to make the code simpler for readers. Apart from that, the implementations look great and I think this will be a great piece of work along the line. LGTM!

jessicajacelyn committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-Gradle

jessicajacelyn committed:
Link project with Gradle and added checkstyle support

snoidetx commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+        super(desc, "D");
+        this.by = by;
+    }
+    public Deadline(String desc, LocalDateTime by, boolean done) {
```
I think it would be better to leave a line between each two methods.

snoidetx commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,148 @@
+public class Parser {
+    public String[] words; // user input split by spaces to get words
+
+    public Parser(String command, TaskList tasks, Storage storage) throws IOException {
```
This method is a bit long - imagine if you have more commands to be added to the app, then it will be even longer. I would suggest that you can further abstract the implementation of each command into classes or methods.

snoidetx commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,105 @@
+        BufferedReader readFile
+                = new BufferedReader(new FileReader(this.file));
+        ArrayList<Task> tasks = new ArrayList<>();
+        StringBuilder sb;
```
I like the use of StringBuilder! It will improve the efficiency of the code!

snoidetx commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,48 @@
+
+    public boolean isDone;
+    public String desc;
+    public String type;
```
Is it necessary to maintain a type attribute? If it is necessary in your implementation, you might want to consider using `enum` instead of strings as the value of type when you instantiate events/deadlines/... because that will make it more understandable :)

snoidetx commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,14 @@
+package duke;
+
+public class Ui {
```
Since this class is a user interface, you might consider allocate all the input/output tasks to it (this might also correspond to my previous comment about functional abstraction).

snoidetx commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,14 @@
+
+    }
+
+    public void showLoadingError() {
```
For this kind of message, you may want to define them as constants so that it might be easier when you want to modify it or reuse it somewhere else.

snoidetx reviewed (commented):
Overall, I think this is a great programme! You have successfully abstracted the tasks into several classes and methods. Meanwhile, it could be even better if you can further abstract them and use more OOPs! Jiayous!

jessicajacelyn committed:
Complete Level-10

jessicajacelyn committed:
Add assert features

jessicajacelyn committed:
Update GUI appearance and fix bugs

jessicajacelyn committed:
Delete .class directory

jessicajacelyn committed:
Bug fix on 'bye' command

jessicajacelyn committed:
Add representative screenshot of the product

jessicajacelyn committed:
Set theme jekyll-theme-minimal

jessicajacelyn committed:
Update README.md

jessicajacelyn committed:
Update README.md

jessicajacelyn committed:
Bug fixes

jessicajacelyn committed:
Merge branch 'master' of https://github.com/jessicajacelyn/ip

jessicajacelyn committed:
Fix new file creation bug when running .jar

jessicajacelyn committed:
Update README.md

jessicajacelyn committed:
Update README.md

jessicajacelyn committed:
Add a new extension to jBot.

User can now type in "today" to retrieve tasks they have today.

A simple keyword "today" is used for it to be easily remembered and
quickly typed in when user are in a rush.

Date on the saved deadlines and events are compared with today's date and
then listed out as a list.

jessicajacelyn committed:
Merge pull request #2 from jessicajacelyn/b-reminder

Add a new extension to jBot.

jessicajacelyn committed:
Update README.md

jessicajacelyn committed:
Update README.md

Update the release link to download the newest jar file

jessicajacelyn committed:
Add more javaDocs

jessicajacelyn committed:
Merge pull request #4 from jessicajacelyn/b-reminder

Add more javaDocs
